proposals/p3833.md

```diff
@@ -112,6 +113,34 @@ understand how the design works in complex cases and to debug unexpected
 behavior in the toolchain. This means starting with that implementation doesn't
 result in sunk cost.
 
+### Example: overloaded operators
+
+As an example to help illustrate this, how should the toolchain model overloaded
+operators? This proposal suggests that _initially_, the implementation should
+aim to fully model the rewrite-based dispatch through interfaces in the prelude.
+That is, each use af `x + y` should turn into roughly equivalent SemIR as would
+be used to model the rewritten semantics of `x.(Core.AddWith(typeof(y)).Op)(y)`.
```

Comment:
@zygoloid Is this syntax allowed?

Reply:
Roughly, yes, I think so. I see a couple of issues here:

- `typeof` doesn't exist (yet).
- I would not expect us to need `Core` to be in scope -- we mean "the `AddWith` that lives in the prelude" rather than "the result of name lookup into the name `Core`" here. In #1083, the rule was that we look for an implementation of the interface from the standard library, not that we do a name lookup to find the name `Core` and look for these various names within whatever we find.

Reply:
I think the syntax is now clarified as imagined syntax and not doing weird things with name lookup.